eng/common/tools.sh

```diff
@@ -165,18 +165,19 @@ function InstallDotNet {
   local install_script=$_GetDotNetInstallScript
 
   local archArg=''
-  if [[ "$#" -ge "3" ]]; then
+  if [[ "$#" -ge 3 && "$3" != '' ]]; then
```

Comment:
Another way to do this in bash is like this. Consider updating for consistency with the rest of the code in this file.

```suggestion
  if [[ -n "${3:-}" ]]; then
```